### ISSUE TITLE
Less aggressive masking for first echo

### ISSUE THREAD

**Issue by handwerkerd** [ME-ICA/tedana#827]
### Summary

There are several anecdotes where CSF and even white matter voxels are being masked. This isn't necessarily a problem for tedana denoising, but may affect how tedana interacts with other pipelines.  (For example: https://github.com/NBCLab/power-replication/issues/14 )
 
These voxels probably shouldn't be masked. Tedana masking is fairly simple and focuses on voxels with relatively lower raw magnitudes. We could see if there is a way to still mask out dropout regions while retaining the ventricles.

### Next Steps

- Agree this is something worth addressing
- Look at other masking methods to see if there's an existing light-weight method

**Comment by tsalo** [comment]
What I ended up doing in the fork I'm using for that analysis is to assume that all voxels in any user-provided mask has good data in at least the first two echoes. It probably means that we end up relying on our T2* value restrictions more in those voxels, which is unfortunate, but given how we perform optimal combination, it shouldn't be a big problem.

**Comment by handwerkerd** [comment]
For your fork, what if you assume the first echo only is good and you'd remove voxels that drop too much from the first to second echo? Right now, we're independently creating a mask for each echo. In the revised version, you'd find the median value for each echo & get median_pchange = (median_echo1-median_echo2)/(median echo1). If pchange in a voxel > X*median_pchange (say for X=2) then assuming you're seeing significant dropout rather than a decay curve and mask that voxel.

This is something I could see playing with in tedana proper, but I'm not sure how to prioritize. I originally set this as medium priority, but I just shifted to low priority. Do you think this would be better to address sooner rather than later or is it more of an edge case?

**Comment by tsalo** [comment]
>For your fork, what if you assume the first echo only is good and you'd remove voxels that drop too much from the first to second echo?

My biggest issue in my analysis was that CSF and WM voxels were being removed, which prevented me from aCompCor. Dropping _any_ voxels from the fMRIPrep brain mask would have been a problem.

>In the revised version, you'd find the median value for each echo & get median_pchange = (median_echo1-median_echo2)/(median echo1). If pchange in a voxel > X*median_pchange (say for X=2) then assuming you're seeing significant dropout rather than a decay curve and mask that voxel.

This sounds similar to #312. I think looking at the signal decay instead of each echo separately could be a great improvement for the adaptive mask, but I think we'd also still end up with issues in the CSF and WM, since I've seen signal curves that don't actually decrease regularly there. Perhaps the two-compartment model (#212) would solve that issue, but I believe that would require a lot of echoes. At the moment, just assuming that all voxels in the brain mask are _at least_ good enough for optimal combination (and leaning heavily on our T2* restrictions) seems to work well.

**Comment by tsalo** [comment]
I like the idea of region-filling using the T2* and S0 estimates.

**Comment by tsalo** [comment]
In one of our recent meetings we talked about just adding a warning if users don't provide their own brain mask. I think we can just do that and not worry about improving on `compute_epi_mask`.

**Comment by handwerkerd** [comment]
I agree. That's already merged in #1203 so I think we can close this issue.
